misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/CuratorFrameworkProvider.kt

```diff
@@ -34,15 +36,16 @@ internal class CuratorFrameworkProvider @Inject internal constructor(
             .setNameFormat("zk-clustering-${config.zk_connect}")
             .build())
         .zookeeperFactory { connectString, sessionTimeout, watcher, canBeReadOnly ->
-          val zk = DefaultZookeeperFactory().newZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly)
+          val clientConfig = ZKClientConfig()
           if (tlsEnabled) {
-            zk.clientConfig.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty")
-            zk.clientConfig.setProperty("zookeeper.client.secure", "true")
-            zk.clientConfig.setProperty("zookeeper.ssl.keyStore.location", config.cert_store?.resource)
-            zk.clientConfig.setProperty("zookeeper.ssl.keyStore.password", config.cert_store?.passphrase)
-            zk.clientConfig.setProperty("zookeeper.ssl.trustStore.location", config.trust_store?.resource)
-            zk.clientConfig.setProperty("zookeeper.ssl.trustStore.password", config.trust_store?.passphrase)
+            clientConfig.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty")
+            clientConfig.setProperty("zookeeper.client.secure", "true")
+            clientConfig.setProperty("zookeeper.ssl.keyStore.location", config.cert_store?.resource)
+            clientConfig.setProperty("zookeeper.ssl.keyStore.password", config.cert_store?.passphrase)
+            clientConfig.setProperty("zookeeper.ssl.trustStore.location", config.trust_store?.resource)
+            clientConfig.setProperty("zookeeper.ssl.trustStore.password", config.trust_store?.passphrase)
           }
+          val zk = ZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly, clientConfig)
           zk
```

Comment:
is zk redundant now and can we just inline the constructed object?

Reply:
lol yes. Will fix up